### PR TITLE
Trapezoidal trajectory generator (WIP)

### PR DESCRIPTION
**Pull request by angstrem98**
This is a generator for trajectories with trapezoidal velocity shape (bang-zero-bang control on acceleration level).
It is possible to generate a trajectory with non-zero final velocity (useful for chaining moves).
Interrupting current generation is possible by setting a new setpoint, but make sure to pass it current velocity.

Figure shows plots of position, velocity and acceleration. Max velocity and acceleration are 20 and 2, respectively.
First setpoint is -300. In time instant 10, new setpoint is commanded: +400. Velocity drops to zero and movement continues towards +400
Figure_1
.

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
Ja sam mislio da je ovo ROS cvor. Ja bih da napravimo ROS cvor za kalibraciju odometrije i ovo integrisemo

**Comment by angstrem98** [comment]
Distance angle regulator sa osnovnim funkcijama dodat.

File: mep3_navigation/src/distance_angle/distance_angle_regulator.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+    tf2::Quaternion q(
+        msg->pose.pose.orientation.x,
+        msg->pose.pose.orientation.y,
+        msg->pose.pose.orientation.z,
+        msg->pose.pose.orientation.w);
+    tf2::Matrix3x3 m(q);
+    double roll, pitch, yaw;
+    m.getRPY(roll, pitch, yaw);
+    robot_angle_ = yaw;
```
Comment (lukicdarkoo) [inline mep3_navigation/src/distance_angle/distance_angle_regulator.cpp:86]: S obzirom da imamo rotaciju samo oko z ose (yaw) Euler ugao bi trebalo da je jednak `2 asin (q_z)` (molim te provjeri):
```suggestion
    robot_angle_ = 2 * std::asin(q.z);
```

**Review by lukicdarkoo — CHANGES_REQUESTED** [review]
Prilicno dobro!

Na sastanku, moras mi samo objasniti kako kompenzuje gresku u uglu tokom linearnog kretanja.

File: mep3_navigation/src/distance_angle/distance_angle_regulator.cpp (hunk 1 above)
Comment (angstrem98) [reply mep3_navigation/src/distance_angle/distance_angle_regulator.cpp:106]: Ne moze, menjaju se `z` i `w`

Comment (lukicdarkoo) [reply mep3_navigation/src/distance_angle/distance_angle_regulator.cpp]: Tvoje novo rjesenje je mnogo bolje. @angstrem98 mozes im ostaviti odgovor na ros.answers :)

**Review by lukicdarkoo — APPROVED** [review]
Hvala :)

**Pull request merged**